src/main/scala/com/microsoft/hyperspace/index/IndexLogEntryTags.scala

```diff
@@ -16,12 +16,52 @@
 
 package com.microsoft.hyperspace.index
 
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+
+import com.microsoft.hyperspace.util.HyperspaceConf
+
 object IndexLogEntryTags {
-  // HYBRIDSCAN_REQUIRED indicates if Hybrid Scan is required for this index or not.
+  // HYBRIDSCAN_REQUIRED indicates if Hybrid Scan is required for the index or not.
   val HYBRIDSCAN_REQUIRED: IndexLogEntryTag[Boolean] =
     IndexLogEntryTag[Boolean]("hybridScanRequired")
 
   // COMMON_SOURCE_SIZE_IN_BYTES stores overlapping bytes of index source files and given relation.
   val COMMON_SOURCE_SIZE_IN_BYTES: IndexLogEntryTag[Long] =
     IndexLogEntryTag[Long]("commonSourceSizeInBytes")
+
+  // IS_SIGNATURE_MATCH indicates if the plan has the same signature value with the index or not.
+  val IS_SIGNATURE_MATCH: IndexLogEntryTag[Boolean] =
+    IndexLogEntryTag[Boolean]("isSignatureMatch")
+
+  // IS_HYBRIDSCAN_CANDIDATE indicates if the index can be applied to the plan using Hybrid Scan.
+  val IS_HYBRIDSCAN_CANDIDATE: IndexLogEntryTag[Boolean] =
+    IndexLogEntryTag[Boolean]("hybridScanCandidate")
+
+  // IS_HYBRIDSCAN_CONFIG_CAPTURE contains values of Hybrid Scan related configs.
+  val HYBRIDSCAN_CONFIG_CAPTURE: IndexLogEntryTag[Seq[String]] =
+    IndexLogEntryTag[Seq[String]]("hybridScanConfigCapture")
+
+  private def getHybridScanConfigs(spark: SparkSession): Seq[String] = {
+    Seq(
+      HyperspaceConf.hybridScanEnabled(spark).toString,
+      HyperspaceConf.hybridScanDeletedRatioThreshold(spark).toString,
+      HyperspaceConf.hybridScanAppendedRatioThreshold(spark).toString)
+  }
+
+  private[hyperspace] def resetHybridScanTagsIfNeeded(
```

Comment:
Should we move this logic out of `IndexLogEntryTags` which should just define tags or tag related functionality without any business logics? We can move this to where it's used.

Reply:
This seems to be also unsetting `IS_HYBRIDSCAN_CANDIDATE`. Maybe we can rename this to something like `prepareHybridScanCandidateSelection`, etc.